### ISSUE TITLE
 GridsterPush is not working properly 

### ISSUE THREAD

**Issue by danielloynaz** [tiberiuzuld/angular-gridster2#576]
GridsterPush do not work in sometimes, if larger card tries to push smaller card.

This is the code that I'm using to push the cards after a card's rows changed

```typescript
 // Create GridsterPush instance in order to move effected items
const push = new GridsterPush(widgetComponent);

if (push.pushItems(push.fromNorth)) { // push items from a direction

    push.checkPushBack(); // check for items can restore to original position
    push.setPushedItems(); // save affected items
    widgetComponent.setSize();
    widgetComponent.checkItemChanges(widgetComponent.$item, widgetComponent.item);
    push.destroy(); // destroy push instance

} else {

    push.restoreItems(); // restore the initial state before push
    push.destroy(); // destroy push instance

 }  
```

![image](https://user-images.githubusercontent.com/10605297/71482990-84558000-27d3-11ea-93ba-ea4abf53b46f.png)

    
Angular CLI: 8.3.20
Node: 10.16.0
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.803.20
@angular-devkit/build-angular      0.803.20
@angular-devkit/build-optimizer    0.803.20
@angular-devkit/build-webpack      0.803.20
@angular-devkit/core               8.3.20
@angular-devkit/schematics         8.3.20
@angular/cdk                       8.2.3
@angular/cli                       8.3.20
@angular/flex-layout               8.0.0-beta.27
@angular/material                  8.2.3
@angular/material-moment-adapter   8.2.3
@ngtools/webpack                   8.3.20
@schematics/angular                8.3.20
@schematics/update                 0.803.20
rxjs                               6.5.3
typescript                         3.5.3
webpack                            4.39.2

**Comment by lifeinchords** [comment]
Not sure if this is related, I'm running into the reverse- swapping a small item with a larger item doesn't work

**Comment by tiberiuzuld** [comment]
The issue should be fixed now in v10.1.2
